components/script/dom/gamepad.rs

```diff
@@ -75,27 +74,27 @@ impl Gamepad {
                        state: &WebVRGamepadState) -> Root<Gamepad> {
         let buttons = GamepadButtonList::new_from_vr(&global, &state.buttons);
         let pose = VRPose::new(&global, &state.pose);
+
+        let root = reflect_dom_object(box Gamepad::new_inherited(state.gamepad_id,
+                                                                 data.name.clone(),
+                                                                 index,
+                                                                 state.connected,
+                                                                 state.timestamp,
+                                                                 "".into(),
+                                                                 &buttons,
+                                                                 Some(&pose),
+                                                                 data.hand.clone(),
+                                                                 data.display_id),
+                                          global,
+                                          GamepadBinding::Wrap);
         let cx = global.get_cx();
-        rooted!(in (cx) let mut axes = ptr::null_mut());
+        rooted!(in (cx) let mut array = ptr::null_mut());
         unsafe {
-            let _ = Float64Array::create(cx,
-                                         CreateWith::Slice(&state.axes),
-                                         axes.handle_mut());
+           let _ = Float64Array::create(cx, CreateWith::Slice(&state.axes), array.handle_mut());
```

Comment:
Is there a way to have `Float64Array::create` allocate directly into `root.axes` rather than into a dummy variable? @jdm?